src/hooks/useTodos.ts

```diff
@@ -0,0 +1,44 @@
+import { useState } from 'react';
+import { Filter } from '../App';
+import { useDispatch, useSelector } from 'react-redux';
+import { RootState } from '../app/store';
+import { filterSlice } from '../features/filter';
+
+export const useTodos = () => {
+  const dispatch = useDispatch();
+  const [isTodosLoaded, setIsTodosLoaded] = useState(false);
+
+  const [query, setQuery] = useState('');
+
+  const handleResetQuery = () => {
+    setQuery('');
+    dispatch(filterSlice.actions.settingQuery(''));
```

Comment:
Ensure that the `settingQuery` action in `filterSlice` is correctly defined to handle resetting the query. This action should update the state with an empty query string.